artifact/gem5art/artifact/_artifactdb.py

```diff
@@ -198,12 +200,167 @@ def searchByLikeNameType(self, name: str, typ: str, limit: int) -> Iterable[Dict
         for d in data:
             yield d
 
+class ArtifactFileDB(ArtifactDB):
+    """
+    This is a file-based database where Artifacts (as defined in artifacts.py)
+    are stored in a JSON file. This database does not keep track of files.
+
+    This database stores the data in three collections:
+    - artifacts: This stores the json serialized Artifact class
+    """
+
+    class ArtifactEncoder(json.JSONEncoder):
+        def default(self, obj):
+            print(type(obj))
+            if isinstance(obj, UUID):
+                return str(obj)
+            return ArtifactFileDB.ArtifactEncoder(self, obj)
+
+    _json_file: Path
+    _uuid_artifact_map: Dict[str, Dict[str,str]]
+    _hash_uuid_map: Dict[str, List[str]]
+
+    def __init__(self, uri :str) -> None:
+        """Initialize the mongodb connection and grab pointers to the databases
+           uri is the location of the database in a mongodb compatible form.
+           http://dochub.mongodb.org/core/connections.
+        """
+        filepath = uri.split("file://")[1]
```

Comment:
Using https://docs.python.org/3/library/urllib.parse.html#url-parsing would be better